### PR TITLE
Add the comparator store

### PR DESCRIPTION
**Pull request by jonathan-best**
Add comparator store and actions to the Tag comparator

### CODE REVIEW

File: src/server/routers/api-routes.js [hunk 1]
```diff
@@ -12,21 +12,28 @@ import SearchDataFormatter from '../formatters/Search';
 let router = express.Router();
 router.use(bodyParser.json());
 
-router.get('/:category(articles|topics|authors)/:uuid', category);
-router.post('/:category(articles|topics|authors)/:uuid', category);
+router.get('/:category(articles|topics|authors)/:uuid([a-zA-Z0-9]{0,10}-[a-zA-Z0-9]{0,10}-[a-zA-Z0-9]{0,10}-[a-zA-Z0-9]{0,10}-[a-zA-Z0-9]{0,15})', getAPIData); //delete me use postman
```
Comment (peter-mouland) [inline src/server/routers/api-routes.js]: this is a pretty sweet regex for UUIDs
```
[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}
```

File: src/shared/stores/ComparatorStore.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  constructor() {
+    this.data = null;
+    this.errorMessage = null;
+    this.loading = false;
+    this.bindListeners({
+      handleUpdateData: ComparatorActions.UPDATE_DATA,
+      handleLoadingData: ComparatorActions.LOADING_DATA,
+      handleLoadingFailed: ComparatorActions.LOADING_FAILED,
+      handleDestroy: ComparatorActions.DESTROY
+    });
```
Comment (peter-mouland) [inline src/shared/stores/ComparatorStore.js]: the 6 lines of `this.bindListeners({...})` can be replaced with 
```
    this.bindActions(ComparatorActions);
```

**Comment by radiodario** [comment]
The code looks good - do we need to activate the comparator bar?

File: src/server/routers/api-routes.js [hunk 3]
```diff
@@ -36,6 +38,7 @@ function category(req, res, next) {
         .catch((error) => {
           switch (error.name) {
             case 'ArticleNotFoundError':
+            case 'ComparatorNotFoundError':
```
Comment (peter-mouland) [inline src/server/routers/api-routes.js]: don't need this line

**Comment by peter-mouland** [comment]
cool turn in prod config and good to go

**Pull request merged**